src/App.tsx

```diff
@@ -1,223 +1,224 @@
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { ErrorNotification } from
+  './components/ErrorNotification/ErrorNotification';
+import { Todo } from './types/Todo';
+import {
+  addTodo,
+  getTodos,
+  removeTodo,
+  updateTodo,
+} from './api/todos';
+import { FilterType } from './types/FiltersType';
+import { getFilteredTodos } from './components/helpers/getFilteredTodos';
+import { TodoItem } from './components/TodoItem/TodoItem';
+
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [completedFilter, setCompletedFilter] = useState(FilterType.All);
+  const [errorText, setErrorText] = useState('');
+
+  const [newTitle, setNewTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [selectedTodosIds, setSelectedTodosIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => setErrorText('Unable to load a todos'));
+    }
+  }, [user?.id]);
+
+  const addNewTodo = useCallback((event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!newTitle) {
+      setErrorText('Title can\'t be empty');
+
+      return;
+    }
+
+    const newTodoAdding = async () => {
+      setIsAdding(true);
+
+      if (user) {
+        setTempTodo({
+          id: 0,
+          userId: user?.id,
+          title: newTitle,
+          completed: false,
+        });
+
+        try {
+          const newTodo = await addTodo({
+            userId: user.id,
+            title: newTitle,
+            completed: false,
+          });
+
+          setTempTodo(null);
+
+          setTodos(currentTodos => [...currentTodos, newTodo]);
+
+          setNewTitle('');
+        } catch (e) {
+          setErrorText('Unable to add a todo');
+        } finally {
+          setIsAdding(false);
+        }
+      }
+    };
+
+    newTodoAdding();
+  }, [newTitle]);
+
+  const deleteTodo = useCallback(async (todoId: number) => {
+    setSelectedTodosIds(prev => [...prev, todoId]);
+    try {
+      await removeTodo(todoId);
+
+      setTodos(currentTodos => currentTodos.filter(
+        todo => todo.id !== todoId,
+      ));
+    } catch (e) {
+      setErrorText('Unable to delete a todo');
+    } finally {
+      setSelectedTodosIds(prev => {
+        return prev.filter(deletingId => deletingId !== todoId);
+      });
+    }
+  }, []);
+
+  const deleteAllComplitedTodos = useCallback(async () => {
+    const completedTodoId = todos
+      .filter(todo => todo.completed)
```

Comment:
You did this action before, here:
```
 const completedTodosLength = useMemo(() => (
    todos.filter(todo => todo.completed).length
  ), [todos]);
```
Maybe, let's memorize completed todos and use them here ?